libshaderc_spvc/src/spvcir_pass.cc

```diff
@@ -127,13 +194,241 @@ void SpvcIrPass::GenerateSpirvCrossIR(Instruction *inst) {
       break;
     }
 
+    // opcode: 16
+    case SpvOpExecutionMode: {
+      auto &execution = ir_->entry_points[inst->GetSingleWordInOperand(0u)];
+      auto mode =
+          static_cast<SpvExecutionMode>(inst->GetSingleWordInOperand(1u));
+      execution.flags.set(mode);
+
+      switch (mode) {
```

Comment:
There are over 40 execution modes in the SPIR-V spec, so I expect eventually this will be better broken out into a helper function.

... Hm. But looking at the table in the SPIR-V spec, only a few have parameters, and only a subset of those are usable in Shader modules, and even fewer are (for now) usable in WebGPU.

So actually, it's perfectly fine to keep this inline with the rest of the code.